weaviate/collection/grpc.py

```diff
@@ -478,20 +479,41 @@ def __call(self) -> List[SearchResult]:
     def __extract_filters(self, weav_filter: Optional[_Filters]) -> Optional[weaviate_pb2.Filters]:
         if weav_filter is None:
             return None
-        from google.protobuf.timestamp_pb2 import Timestamp
 
         if isinstance(weav_filter, _FilterValue):
-            timestamp = Timestamp()
 
-            if isinstance(weav_filter.value, datetime.date):
-                timestamp.FromDatetime(weav_filter.value)
+            def date_and_uuid_and_string_to_text(
+                string_or_date: Union[str, TIME, uuid_lib.UUID]
+            ) -> str:
+                if isinstance(string_or_date, str):
+                    return string_or_date
+
+                if isinstance(string_or_date, uuid_lib.UUID):
+                    return str(string_or_date)
+
+                return _datetime_to_string(string_or_date)
+
+            def date_and_string_list_to_text_array(
+                string_or_date: Union[List[str], List[TIME], List[uuid_lib.UUID]]
+            ) -> List[str]:
+                if isinstance(string_or_date[0], str):
+                    return cast(List[str], string_or_date)
+
+                if isinstance(string_or_date[0], uuid_lib.UUID):
+                    return cast(List[str], [str(uid) for uid in string_or_date])
+
+                dates = cast(List[TIME], string_or_date)
+                return [_datetime_to_string(date) for date in dates]
 
             return weaviate_pb2.Filters(
                 operator=weav_filter.operator,
-                value_text=weav_filter.value if isinstance(weav_filter.value, str) else None,
+                value_text=date_and_uuid_and_string_to_text(weav_filter.value)
+                if isinstance(weav_filter.value, TIME)
+                or isinstance(weav_filter.value, str)
+                or isinstance(weav_filter.value, uuid_lib.UUID)
+                else None,
```

Comment:
I find reading this code needs a decent amount of concentration to extract its action and the movement of all the types, the same goes for L526 to L536 below

Could the logic be contained into single functions for each argument to `weaviate_pb2.Filters`? Perhaps as:
```python
weaviate_pb2.Filters(
    ...
    value_text=self.__parse_filter_value_to_text(weav_filter.value),
    value_text_array=self.__parse_filter_value_to_text_array(weav_filter.value),
    ...
)
```

Reply:
extracted it to an extra class, let me know what you think

Reply:
Looks great, thanks for the effort! 😁